applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.h

```diff
@@ -50,11 +50,227 @@ namespace MPMParticleGeneratorUtility
      * @brief Construct material points or particles from given initial mesh
      * @details Generating particles using a designated shape functions
      */
+    template<SizeType TDimension>
     void GenerateMaterialPointElement(  ModelPart& rBackgroundGridModelPart,
                                         ModelPart& rInitialModelPart,
                                         ModelPart& rMPMModelPart,
-                                        bool IsAxisSymmetry = false,
-                                        bool IsMixedFormulation = false);
+                                        bool IsMixedFormulation = false) {
```

Comment:
Any reason why moving it to the header?

Reply:
It is templated by the dimension now. Can't do this in the cpp.

Reply:
usually in this case it is still implemented in the header and then instantiated
check here: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/processes/apply_ray_casting_process.cpp#L484-L485

Reply:
Thanks. I can maybe do that in another PR. This function is anyways not the most beautiful thing on earth so maybe it will anyways become cleaner in a certain step.